pkg/cmd/repo/create/create.go

```diff
@@ -73,6 +73,15 @@ func NewCmdCreate(f *cmdutil.Factory, runF func(*CreateOptions) error) *cobra.Co
 				opts.Name = args[0]
 			}
 
+			if !opts.IO.CanPrompt() {
+				if opts.Name == "" {
+					return &cmdutil.FlagError{Err: errors.New("must pass name argument when prompts disabled")}
+				}
+				if !opts.Internal && !opts.Private && !opts.Public {
+					opts.Public = true
```

Comment:
I strongly feel that we should either default to `--private` or error out in non-interactive mode when repo visibility wasn't explicitly declared.

Reply:
I set it to error now, happy to update to something else later. I figured the annoyance of getting an error is better than the annoyance of having to go to the browser and toggle the repo visibility if the default isn't what you wanted.